src/input.rs

```diff
@@ -373,6 +421,26 @@ macro_rules! go_extra {
     };
 }
 
+impl<'a, 'b, T, I, E, S> Parser<'a, I, E, S> for &'b T
```

Comment:
These two lifetimes were leftover from me trying something, I don't *think* they're necessary and can remove them if you want.